client/verta/verta/runtime.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+import json
+from typing import Any, Dict
+
+
+class LogHandler:
+    """
+    Container class for collecting and dumping log entries.
+    """
+    def __init__(self):
+        self.attributes: Dict[str, Any] = dict()
+
+    def __repr__(self):
+        return str(self.attributes)
+
+    def add(self, **kwargs):
+        # Overwrites existing key-values
+        self.attributes.update(**kwargs)
+
+    def wipe_clean(self):
+        self.attributes = dict()
+
+# Use a single handler object that is wiped clean after each log entry
+# to minimize memory usage
+LOG_HANDLER = LogHandler()
+
+
+def log(**kwargs) -> None:
+    """
+    Log any number of key-value pairs to the current prediction context.
+    Values are deleted upon completion of the active model's predict() method.
+
+    Parameters
+    ----------
+    kwargs : Any
+        Arguments and values to be added as key-value pairs to the current log record.
+        Values must be serializable to JSON.
+
+    Returns
+    -------
+    None
+    """
+    LOG_HANDLER.add(**kwargs)
+
+
+class ContextLogger():
+    """
+    Context manager for aggregating key-value pairs into a custom log entry.
+    Any logged data is deleted upon exiting the context.
+
+    Attributes
+    ----------
+    as_json : str
+        A JSON string representation of all the current saved logging context.
+    """
+    def __init__(self):
+        self.context = LOG_HANDLER
+        if bool(self.context.attributes): # if not empty dict
```

Comment:
```suggestion
        if self.context.attributes:
```

As far as I know, explicitly casting to `bool` is extremely unidiomatic in Python, and it's almost always clear from context (and better: from type analysis tooling) what's being checked.

Reply:
This has been refactored out.